legal-api/src/legal_api/services/authz.py

```diff
@@ -488,17 +488,17 @@ def is_allowed(business: Business,
                legal_type: str,
                jwt: JwtManager,
                sub_filing_type: str = None,
-               filing_id: int = None):
+               filing: Filing = None):
     """Is allowed to do filing."""
     is_ignore_draft_blockers = False
 
-    if filing_id:
-        if filing := Filing.find_by_id(filing_id):
-            if filing.status not in [Filing.Status.DRAFT.value, Filing.Status.CHANGE_REQUESTED.value]:
-                return False  # common for all filings
-
-            if filing.status == Filing.Status.DRAFT.value:
-                is_ignore_draft_blockers = True
+    if filing:
+        if filing.status not in [Filing.Status.DRAFT.value,
+                                 Filing.Status.CHANGE_REQUESTED.value,
+                                 Filing.Status.APPROVED.value]:
+            return False  # common for all filings
```

Comment:
Why is AWAITING_REVIEW not in this list?

Reply:
When a filing is in `AWAITING_REVIEW` user cannot modify it

Reply:
OK, makes sense. PS I love code comments :)